proof_of_play.go

```diff
@@ -158,3 +176,44 @@ func (p *proofOfPlay) expire(ad Ad) error {
 		return false, nil
 	})
 }
+
+func (p *proofOfPlay) isLeaseExpired(ad Ad) bool {
+	exp, ok := ad["lease_expiry"].(float64)
+	if !ok {
+		return true
+	}
+
+	expiry := time.Unix(int64(exp), 0)
+	return time.Now().After(expiry)
+}
+
+func (p *proofOfPlay) processRetries() {
+	for req := range p.retryQueue {
+		p.retryPoP(req)
+	}
+}
+
+func (p *proofOfPlay) retryPoP(req *PoPRequest) {
+	ad := req.Ad
+
+	if p.isLeaseExpired(ad) {
+		popType := p.getPoPType(req)
+		p.publishEvent(
+			fmt.Sprintf("ad-%s-already-expired", popType),
+			fmt.Sprintf("ad: %s, expiry: %d", ad["id"], ad["lease_expiry"]),
+			"critical")
+		return
+	}
+
+	sleepDuration := RetryInterval - time.Since(req.RequestTime)
+	time.Sleep(sleepDuration)
+	p.requests <- req
+}
+
+func (p proofOfPlay) getPoPType(req *PoPRequest) string {
```

Comment:
very minor, this method could be written as:

if req.Status {
 return "pop"
}

return "expire"